testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/modules/AddRemoveModuleTest.java

```diff
@@ -0,0 +1,120 @@
+import org.wildfly.extras.creaper.core.ManagementClient;
+import org.wildfly.extras.creaper.core.online.CliException;
+import org.wildfly.extras.creaper.core.online.ModelNodeResult;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.OnlineOptions;
+import org.xml.sax.SAXException;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.wildfly.extras.creaper.XmlAssert.assertXmlIdentical;
+
+@RunWith(Arquillian.class)
+public class AddRemoveModuleTest {
+    private static final String TEST_MODULE_NAME = "org.jboss.testmodule";
+    private static final String EXPECTED_MODULE_XML = ""
+            + "<?xml version=\"1.0\" ?>\n"
+            + "\n"
+            + "<module xmlns=\"urn:jboss:module:1.1\" name=\"" + TEST_MODULE_NAME + "\" slot=\"main\">\n"
+            + "\n"
+            + "    <properties>\n"
+            + "        <property name=\"foo\" value=\"bar\"/>\n"
+            + "        <property name=\"john\" value=\"doe\"/>\n"
+            + "    </properties>\n"
+            + "\n"
+            + "    <main-class value=\"test.mainclass\"/>\n"
+            + "\n"
+            + "    <resources>\n"
+            + "        <resource-root path=\"testJar.jar\"/>\n"
+            + "    </resources>\n"
+            + "\n"
+            + "    <dependencies>\n"
+            + "        <module name=\"org.hibernate\"/>\n"
+            + "        <module name=\"org.jboss.as.controller\"/>\n"
+            + "    </dependencies>\n"
+            + "</module>";
+
+    private OnlineManagementClient client;
+
+    @Rule
+    public TemporaryFolder tmp = new TemporaryFolder();
+
+    @Before
+    public void connect() throws IOException {
+        client = ManagementClient.online(OnlineOptions.standalone().localDefault().build());
+    }
+
+    @After
+    public void close() throws IOException {
+        client.close();
+    }
+
+    @Test
+    public void addRemoveModuleTest() throws IOException, CommandFailedException, CliException, SAXException {
+        // create test module
+        File testJar = createTestJar();
+        AddModule addModule = new AddModule.Builder(TEST_MODULE_NAME)
+                .resource(testJar)
+                .resourceDelimiter(":")
+                .dependency("org.hibernate")
+                .dependency("org.jboss.as.controller")
+                .mainClass("test.mainclass")
+                .property("foo", "bar")
+                .property("john", "doe")
+                .build();
+        client.apply(addModule);
+
+        // verify that module was added
+        File asRoot = new File(getPathToAs());
+
+        File module = new File(asRoot, "modules" + File.separator + TEST_MODULE_NAME.replaceAll("\\.", File.separator));
+        assertTrue("Module " + module.getAbsolutePath() + " doesn't exist on path", module.exists());
```

Comment:
I always found this style of assertion failure messages confusing. I'd prefer the "should" style; in this case, `Module ... should exist on path`.